src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,197 @@
+package parser;
+
+import commands.Command;
+import commands.PrintCommand;
+import commands.CommandManager;
+import commands.AddCommand;
+import commands.DeleteCommand;
+import commands.DoneCommand;
+import commands.ExitCommand;
+import commands.FindCommand;
+import commands.UndoableCommand;
+import java.io.IOException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Todo;
+import tasks.Task;
+import util.TaskList;
+import storage.Storage;
+import ui.Ui;
+
+
+import java.text.ParseException;
+
+/**
+ * Encapsulates a handler that tries to make sense of user commands.
+ *
+ *  @author atharvjoshi
+ *  @version CS2103 AY19/20 Sem 1 iP Week 4
+ */
+public class Parser {
+    /**
+     * Enumeration of all command types supported by Duke.
+     */
+    private enum Imperative {
+        BYE, DEADLINE, DELETE, DONE, EVENT, FIND, LIST, TODO, UNDO
```

Comment:
consider declaring a value for your enums i.e. BYE ("bye"), DEADLINE ("deadline")
This prevents the scenario where a user can provide 0, 1, 2 - currently your enums are defaulted to 0, 1 and 2.